engine: add 20.10.8 release notes (#13278)
docker/docker.github.io: engine/release-notes/index.md

# Version 20.10

## 20.10.8
2021-08-03

### Deprecation

- Deprecate support for encrypted TLS private keys.

>   Legacy PEM encryption as specified in RFC 1423 is insecure by design. Since

Reviewer: ```suggestion
  Legacy PEM encryption as specified in RFC 1423 is insecure by design. Because
```

>   it does not authenticate the ciphertext, it is vulnerable to padding oracle

Reviewer: ```suggestion
  it does not authenticate the ciphertext, it is vulnerable to padding Oracle
```

Author: I think `oracle` here is "oracle" (not the company/product). I recall I copied this text from Golang's description of the issue (Golang deprecated this functionality).

Reviewer: Aah, makes sense. Thanks for clarifying.

  attacks that can let an attacker recover the plaintext.

>   Support for this feature is now marked deprecated, and will be removed in an

Reviewer: ```suggestion
  Support for encrypted TLS private keys is now marked deprecated, and will be removed in an
```

  upcoming release. [docker/cli#3219](https://github.com/docker/cli/pull/3219)
- Deprecate Kubernetes stack support
  Following the deprecation of [Compose on Kubernetes](https://github.com/docker/compose-on-kubernetes),

>   support for Kubernetes in the `stack` and `context` commands in the docker CLI

Reviewer: ```suggestion
  support for Kubernetes in the `stack` and `context` commands in the Docker CLI
```

  is now marked as deprecated, and will be removed in an upcoming release [docker/cli#3174](https://github.com/docker/cli/pull/3174).

### Client

> - Fix "Invalid standard handle identifier" errors on Windows [docker/cli#3132](https://github.com/docker/cli/pull/3132).

Reviewer: ```suggestion
- Fix `Invalid standard handle identifier` errors on Windows [docker/cli#3132](https://github.com/docker/cli/pull/3132).
```

### Rootless

> - `dockerd-rootless.sh`: avoid "can't open lock file /run/xtables.lock: Permission denied" error  on SELinux hosts [moby/moby#42462](https://github.com/moby/moby/pull/42462).

Reviewer: ```suggestion
- `dockerd-rootless.sh`: avoid`can't open lock file /run/xtables.lock: Permission denied` error  on SELinux hosts [moby/moby#42462](https://github.com/moby/moby/pull/42462).
```

- disable overlay2 when running with SELinux to prevent permission denied errors [moby/moby#42462](https://github.com/moby/moby/pull/42462).
- fix "x509: certificate signed by unknown authority" error on openSUSE Tumbleweed [moby/moby#42462](https://github.com/moby/moby/pull/42462).

### Runtime

- Print a warning when using the `--platform` option to pull a single-arch image
  that does not match the host's native architecture [moby/moby#42633](https://github.com/moby/moby/pull/42633).

> - Fix incorrect "Your kernel does not support swap memory limit" warning when

Reviewer: ```suggestion
- Fix the incorrect `Your kernel does not support swap memory limit` warning when
```
